src/ast/astBuilder.ts

```diff
@@ -377,22 +427,101 @@ export class ASTBuilder implements CVisitor<any> {
     throw new UnreachableCaseError();
   }
 
-  visitForCondition(ctx: ForConditionContext): BaseNode {
-    throw new Error('Method not implemented.');
+  visitForCondition(ctx: ForConditionContext): ForCondition {
+    const forDeclaration = ctx.forDeclaration();
+    const expression = ctx.expression();
+
+    if (forDeclaration !== undefined && expression !== undefined) {
+      throw new BrokenInvariantError(
+        'Encountered a ForCondition with both an init declaration and init expression.'
+      );
+    }
+
+    const init =
+      forDeclaration !== undefined
+        ? this.visitForDeclaration(forDeclaration)
+        : expression !== undefined
+        ? this.visitExpression(expression)
+        : undefined;
+
+    const firstForExpression = ctx.forExpression(0);
+    const test =
+      firstForExpression !== undefined
+        ? this.visitForExpression(firstForExpression)
+        : undefined;
+    const secondForExpression = ctx.forExpression(1);
+    const update =
+      secondForExpression !== undefined
+        ? this.visitForExpression(secondForExpression)
+        : undefined;
+
+    return {
+      type: 'ForCondition',
+      init,
+      test,
+      update
+    };
   }
 
-  visitForDeclaration(ctx: ForDeclarationContext): BaseNode {
-    throw new Error('Method not implemented.');
+  visitForDeclaration(ctx: ForDeclarationContext): VariableDeclaration {
```

Comment:
I copied `visitDeclaration` first. Not sure if `VariableDeclaration` will change in any form based on the todo?

Reply:
`VariableDeclaration` will most likely change once we add all the specifiers. Anyway, TypeScript should flag out any issues when we eventually do so, so not a big deal.